atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -185,22 +185,25 @@ def __init__(self, tasker, workflow, platforms, build_kwargs,
 
         self.worker_builds = []
 
-    def get_excluded_platforms(self):
-        df_dir = self.workflow.source.get_dockerfile_path()[1]
-        exclude_platforms = set()
-        exclude_path = os.path.join(df_dir, self.EXCLUDE_FILENAME)
-        if os.path.exists(exclude_path):
-            with open(exclude_path) as f:
-                for platform in f:
-                    platform = platform.strip()
-                    if not platform:
-                        continue
-                    exclude_platforms.add(platform)
-
-        return exclude_platforms
+    def make_list(self, value):
+        if not isinstance(value, list):
+            value = [value]
+        return value
 
     def get_platforms(self):
-        return self.platforms - self.get_excluded_platforms()
+        df_dir = self.workflow.source.get_dockerfile_path()[1]
+        excluded_platforms = set()
+        container_path = os.path.join(df_dir, self.CONTAINER_FILENAME)
+        if os.path.exists(container_path):
+            with open(container_path) as f:
+                data = yaml.load(f)
+                if data is None or 'platforms' not in data or data['platforms'] is None:
+                    return self.platforms
+                excluded_platforms = set(self.make_list(data['platforms'].get('not', [])))
+                only_platforms = set(self.make_list(data['platforms'].get('only', [])))
+                if only_platforms:
+                    self.platforms = self.platforms & only_platforms
```

Comment:
I don't think you can. If only_platforms is () because no one put an only: field in the yaml, you want to use the default platforms, not no platforms. I thought not: and only: were optional.